src/Core/Services/StateService.cs

```diff
@@ -1001,17 +1001,17 @@ public async Task SetPushInitialPromptShownAsync(bool? value)
             await SetValueAsync(key, value, options);
         }
 
-        public async Task<DateTime?> GetPushLastRegistrationDateAsync()
+        public async Task<DateTime?> GetPushLastRegistrationDateAsync(string userId = null)
         {
-            var options = await GetDefaultStorageOptionsAsync();
-            var key = Constants.PushLastRegistrationDateKey;
+            var options = ReconcileOptions(new StorageOptions { UserId = userId }, await GetDefaultStorageOptionsAsync());
+            var key = Constants.PushLastRegistrationDateKey(userId);
```

Comment:
You have to reference `reconciledOptions.UserId` here otherwise you'll never get the current active userId